identity/backend/src/main/java/io/camunda/identity/usermanagement/service/MembershipService.java

```diff
@@ -31,21 +31,21 @@ public MembershipService(
     this.membershipRepository = membershipRepository;
   }
 
-  public void addUserToGroup(final CamundaUser user, final Group group) {
-    userDetailsManager.addUserToGroup(user.username(), group.name());
+  public void addUserToGroup(final CamundaUser user, final CamundaGroup group) {
+    userDetailsManager.addUserToGroup(user.getUsername(), group.name());
   }
 
-  public void removeUserFromGroup(final CamundaUser user, final Group group) {
-    userDetailsManager.removeUserFromGroup(user.username(), group.name());
+  public void removeUserFromGroup(final CamundaUser user, final CamundaGroup group) {
+    userDetailsManager.removeUserFromGroup(user.getUsername(), group.name());
   }
 
-  public List<CamundaUser> getMembers(final Group group) {
+  public List<CamundaUser> getMembers(final CamundaGroup group) {
     return userDetailsManager.findUsersInGroup(group.name()).stream()
         .map(userService::findUserByUsername)
         .toList();
   }
```

Comment:
I think the above method will execute a select with a join between users and profiles for each user in that group.

Is there a way to pass in all usernames that are returned by `userDetailsManager.findUsersInGroup` and get their details in one potentially faster statement than one by one?